### PR TITLE
updated api docs

### PR DESCRIPTION
**Pull request by ellenevans**
These are the updated api documents for v0.15.0.  The changes appear to have been quite significant - the build took several hours - so please look over them carefully.  Thanks.

### CODE REVIEW

File: docs/reference/serving.md [hunk 1]
```diff
@@ -2642,8 +2642,8 @@ RevisionTemplateSpec
 </h3>
 <p>
 (<em>Appears on:</em>
-<a href="#serving.knative.dev/v1beta1.Configuration">Configuration</a>, 
-<a href="#serving.knative.dev/v1.Configuration">Configuration</a>)
+<a href="#serving.knative.dev/v1.Configuration">Configuration</a>,
```
Comment (mattmoor) [inline docs/reference/serving.md:2645]: Remove trailing whitespace:
```suggestion
<a href="#serving.knative.dev/v1.Configuration">Configuration</a>,
```

File: docs/reference/serving.md [hunk 2]
```diff
@@ -2734,12 +2734,51 @@ Configuration. It might not be ready yet, for that use LatestReadyRevisionName.<
...
+</tr>
+</tbody>
+</table>
 <h3 id="serving.knative.dev/v1.RevisionSpec">RevisionSpec
 </h3>
 <p>
 (<em>Appears on:</em>
-<a href="#serving.knative.dev/v1beta1.Revision">Revision</a>, 
 <a href="#serving.knative.dev/v1.Revision">Revision</a>, 
+<a href="#serving.knative.dev/v1beta1.Revision">Revision</a>,
```
Comment (mattmoor) [inline docs/reference/serving.md:2781]: Remove trailing whitespace:
```suggestion
<a href="#serving.knative.dev/v1beta1.Revision">Revision</a>,
```

File: docs/reference/serving.md [hunk 3]
```diff
@@ -2804,8 +2843,8 @@ be provided.</p>
 </h3>
 <p>
 (<em>Appears on:</em>
-<a href="#serving.knative.dev/v1beta1.Revision">Revision</a>, 
-<a href="#serving.knative.dev/v1.Revision">Revision</a>)
+<a href="#serving.knative.dev/v1.Revision">Revision</a>,
```
Comment (mattmoor) [inline docs/reference/serving.md:2846]: Remove trailing whitespace:
```suggestion
<a href="#serving.knative.dev/v1.Revision">Revision</a>,
```

File: docs/reference/serving.md [hunk 4]
```diff
@@ -2992,8 +3038,8 @@ be provided.</p>
 </h3>
 <p>
 (<em>Appears on:</em>
-<a href="#serving.knative.dev/v1beta1.Route">Route</a>, 
 <a href="#serving.knative.dev/v1.Route">Route</a>, 
+<a href="#serving.knative.dev/v1beta1.Route">Route</a>,
```
Comment (mattmoor) [inline docs/reference/serving.md:3042]: Remove trailing whitespace:
```suggestion
<a href="#serving.knative.dev/v1beta1.Route">Route</a>,
```

File: docs/reference/serving.md [hunk 5]
```diff
@@ -3028,8 +3074,8 @@ revisions and configurations.</p>
 </h3>
 <p>
 (<em>Appears on:</em>
-<a href="#serving.knative.dev/v1beta1.Route">Route</a>, 
-<a href="#serving.knative.dev/v1.Route">Route</a>)
+<a href="#serving.knative.dev/v1.Route">Route</a>,
```
Comment (mattmoor) [inline docs/reference/serving.md:3077]: Remove trailing whitespace:
```suggestion
<a href="#serving.knative.dev/v1.Route">Route</a>,
```

File: docs/reference/serving.md (hunk 1 above)
Comment (mattmoor) [inline docs/reference/serving.md:2645]: Remove trailing whitespace:
```suggestion
<a href="#serving.knative.dev/v1.Configuration">Configuration</a>,
```

File: docs/reference/serving.md (hunk 2 above)
Comment (mattmoor) [inline docs/reference/serving.md:2781]: Remove trailing whitespace:
```suggestion
<a href="#serving.knative.dev/v1beta1.Revision">Revision</a>,
```

File: docs/reference/serving.md (hunk 3 above)
Comment (mattmoor) [inline docs/reference/serving.md:2846]: Remove trailing whitespace:
```suggestion
<a href="#serving.knative.dev/v1.Revision">Revision</a>,
```

File: docs/reference/serving.md (hunk 4 above)
Comment (mattmoor) [inline docs/reference/serving.md:3042]: Remove trailing whitespace:
```suggestion
<a href="#serving.knative.dev/v1beta1.Route">Route</a>,
```

File: docs/reference/serving.md (hunk 5 above)
Comment (mattmoor) [inline docs/reference/serving.md:3077]: Remove trailing whitespace:
```suggestion
<a href="#serving.knative.dev/v1.Route">Route</a>,
```

File: docs/reference/serving.md (hunk 1 above)
Comment (mattmoor) [inline docs/reference/serving.md:2645]: Remove trailing whitespace:
```suggestion
<a href="#serving.knative.dev/v1.Configuration">Configuration</a>,
```

File: docs/reference/serving.md (hunk 2 above)
Comment (mattmoor) [inline docs/reference/serving.md:2781]: Remove trailing whitespace:
```suggestion
<a href="#serving.knative.dev/v1beta1.Revision">Revision</a>,
```

File: docs/reference/serving.md (hunk 3 above)
Comment (mattmoor) [inline docs/reference/serving.md:2846]: Remove trailing whitespace:
```suggestion
<a href="#serving.knative.dev/v1.Revision">Revision</a>,
```

File: docs/reference/serving.md (hunk 4 above)
Comment (mattmoor) [inline docs/reference/serving.md:3042]: Remove trailing whitespace:
```suggestion
<a href="#serving.knative.dev/v1beta1.Route">Route</a>,
```

File: docs/reference/serving.md (hunk 5 above)
Comment (mattmoor) [inline docs/reference/serving.md:3077]: Remove trailing whitespace:
```suggestion
<a href="#serving.knative.dev/v1.Route">Route</a>,
```

File: docs/reference/serving.md (hunk 1 above)
Comment (mattmoor) [inline docs/reference/serving.md:2645]: Remove trailing whitespace:
```suggestion
<a href="#serving.knative.dev/v1.Configuration">Configuration</a>,
```

File: docs/reference/serving.md (hunk 2 above)
Comment (mattmoor) [inline docs/reference/serving.md:2781]: Remove trailing whitespace:
```suggestion
<a href="#serving.knative.dev/v1beta1.Revision">Revision</a>,
```

File: docs/reference/serving.md (hunk 3 above)
Comment (mattmoor) [inline docs/reference/serving.md:2846]: Remove trailing whitespace:
```suggestion
<a href="#serving.knative.dev/v1.Revision">Revision</a>,
```

File: docs/reference/serving.md (hunk 4 above)
Comment (mattmoor) [inline docs/reference/serving.md:3042]: Remove trailing whitespace:
```suggestion
<a href="#serving.knative.dev/v1beta1.Route">Route</a>,
```

File: docs/reference/serving.md (hunk 5 above)
Comment (mattmoor) [inline docs/reference/serving.md:3077]: Remove trailing whitespace:
```suggestion
<a href="#serving.knative.dev/v1.Route">Route</a>,
```

File: docs/reference/serving.md (hunk 1 above)
Comment (mattmoor) [inline docs/reference/serving.md:2645]: Remove trailing whitespace:
```suggestion
<a href="#serving.knative.dev/v1.Configuration">Configuration</a>,
```

File: docs/reference/serving.md (hunk 2 above)
Comment (mattmoor) [inline docs/reference/serving.md:2781]: Remove trailing whitespace:
```suggestion
<a href="#serving.knative.dev/v1beta1.Revision">Revision</a>,
```

File: docs/reference/serving.md (hunk 3 above)
Comment (mattmoor) [inline docs/reference/serving.md:2846]: Remove trailing whitespace:
```suggestion
<a href="#serving.knative.dev/v1.Revision">Revision</a>,
```

File: docs/reference/serving.md (hunk 4 above)
Comment (mattmoor) [inline docs/reference/serving.md:3042]: Remove trailing whitespace:
```suggestion
<a href="#serving.knative.dev/v1beta1.Route">Route</a>,
```

File: docs/reference/serving.md (hunk 5 above)
Comment (mattmoor) [inline docs/reference/serving.md:3077]: Remove trailing whitespace:
```suggestion
<a href="#serving.knative.dev/v1.Route">Route</a>,
```

**Comment by ellenevans** [comment]
/assign mattmoor

Richie is on carers leave and didn't see this.  Sam has moved on.  Neither would be able to evaluate any of this in terms of correct content. I can't release the docs until this is checked in.

**Pull request merged**